libraries/ESP8266WiFiMesh/src/TcpIpMeshBackend.cpp

```diff
@@ -0,0 +1,560 @@
+ 
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include <WiFiClient.h> 
+#include <WiFiServer.h>
+#include <assert.h>
+#include <Schedule.h>
+
+#include "TcpIpMeshBackend.h"
+#include "TypeConversionFunctions.h"
+#include "MutexTracker.h"
+
+#define SERVER_IP_ADDR      "192.168.4.1"
+
+const IPAddress TcpIpMeshBackend::emptyIP = IPAddress();
+
+bool TcpIpMeshBackend::_tcpIpTransmissionMutex = false;
+bool TcpIpMeshBackend::_tcpIpConnectionQueueMutex = false;
+
+String TcpIpMeshBackend::lastSSID = "";
+bool TcpIpMeshBackend::staticIPActivated = false;
+
+String TcpIpMeshBackend::_temporaryMessage = "";
+
+// IP needs to be at the same subnet as server gateway (192.168.4 in this case). Station gateway ip must match ip for server.
+IPAddress TcpIpMeshBackend::staticIP = emptyIP;
+IPAddress TcpIpMeshBackend::gateway = IPAddress(192,168,4,1);
+IPAddress TcpIpMeshBackend::subnetMask = IPAddress(255,255,255,0);
+
+std::vector<TcpIpNetworkInfo> TcpIpMeshBackend::_connectionQueue = {};
+std::vector<TransmissionOutcome> TcpIpMeshBackend::_latestTransmissionOutcomes = {};
+
+TcpIpMeshBackend::TcpIpMeshBackend(requestHandlerType requestHandler, responseHandlerType responseHandler, 
+                                   networkFilterType networkFilter, const String &meshPassword, const String &ssidPrefix, 
+                                   const String &ssidSuffix, bool verboseMode, uint8 meshWiFiChannel, uint16_t serverPort) 
+                                   : MeshBackendBase(requestHandler, responseHandler, networkFilter, MB_TCP_IP), _server(serverPort)
+{
+  setSSID(ssidPrefix, "", ssidSuffix);
+  setMeshPassword(meshPassword);
+  setVerboseModeState(verboseMode);
+  setWiFiChannel(meshWiFiChannel);
+  setServerPort(serverPort);
+}
+
+std::vector<TcpIpNetworkInfo> & TcpIpMeshBackend::connectionQueue()
+{
+  MutexTracker connectionQueueMutexTracker(_tcpIpConnectionQueueMutex);
+  if(!connectionQueueMutexTracker.mutexCaptured())
+  {
+    assert(false && "ERROR! connectionQueue locked. Don't call connectionQueue() from callbacks other than NetworkFilter as this may corrupt program state!"); 
+  }
+  
+  return _connectionQueue;
+}
+
+const std::vector<TcpIpNetworkInfo> & TcpIpMeshBackend::constConnectionQueue()
```

Comment:
Personal nitpick, feel free to disregard:
Prefer putting the name and args on a different line, at least in cases with such long signature:
```cpp
const std::vector<TcpIpNetworkInfo> & 
TcpIpMeshBackend::constConnectionQueue()
```

It improves readability at a glance.